dragonfly/implementations/roach2interface.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""
+"""
+
+
+from __future__ import absolute_import
+
+import logging
+import os
+import adc5g
+import numpy as np
+#import time
+from dripline import core
+from .ethernet_provider import EthernetProvider
+
+
+logger = logging.getLogger(__name__)
+
+#phasmid import
+
+try:
+    from .r2daq import ArtooDaq
+    logger.info('Imported ArtooDaq')
+
+except ImportError:
+    class ArtooDaq(object):
+        def __init__(self, *args, **kwargs):
+            raise RuntimeError("Dependency <ArtooDaq> not found but is required for ROACH2 support.")
+
+
+__all__ = []
+
+
+
+__all__.append('Roach2Provider')
+class Roach2Provider(ArtooDaq, core.Provider):
+    '''
+    A DAQProvider for interacting with the Roach
+    '''
+    def __init__(self, **kwargs):
+
+
+        core.Provider.__init__(self, **kwargs)
+
+
+
+__all__.append('Roach2Interface')
+class Roach2Interface(Roach2Provider, EthernetProvider):
+    def __init__(self,
+                 roach2_hostname = 'led',
+                 source_ip_a = None,
```

Comment:
Sorry I missed this, it would probably have been much cleaner to use a dictionary of configurations for each psyllid target, something like:

```python
psyllid_a_config = {'source_ip': None,
                               ' source_port': None,
                               ...,
                              }
```
And then have the class store all of those in a dictionary too:
```python
psyllid_configs{'a': psyllid_a_config,
                        'b': psyllid_b_config,
                         ...
                        }
```
That way you could take all the places where you treat each channel and instead have one block of code that loops over the channels and uses the same code with `psyllid_configs['a']['source_ip'] etc.